examples/E2E-RAG-App/gradio_interface.py

```diff
@@ -0,0 +1,242 @@
+        self.host = host
+        self.port = port
+        self.docs_dir = docs_dir
+        self.client = LlamaStackClient(base_url=f"http://{host}:{port}")
+        self.chroma_client = chromadb.HttpClient(host=host, port=chroma_port)
+        self.agent = None
+        self.session_id = None
+        self.memory_bank_id = "test_bank_232"
+
+    async def initialize_system(self):
+        """Initialize the entire system including memory bank and agent."""
+        await self.setup_memory_bank()
+        await self.initialize_agent()
+
+    async def setup_memory_bank(self):
+        """Set up the memory bank if it doesn't exist."""
+        providers = self.client.providers.list()
+        provider_id = providers["memory"][0].provider_id
+        collections = self.chroma_client.list_collections()
+
+        if any(col.name == self.memory_bank_id for col in collections):
+            print(f"The collection '{self.memory_bank_id}' exists.")
+        else:
+            print(
+                f"The collection '{self.memory_bank_id}' does not exist. Creating the collection..."
+            )
+            self.client.memory_banks.register(
+                memory_bank_id=self.memory_bank_id,
+                params={
+                    "embedding_model": "all-MiniLM-L6-v2",
+                    "chunk_size_in_tokens": 100,
+                    "overlap_size_in_tokens": 10,
+                },
+                provider_id=provider_id,
+            )
+            await self.load_documents()
+            print(f"Memory bank registered.")
+
+    async def load_documents(self):
+        """Load documents from the specified directory into memory bank."""
+        documents = []
+        for filename in os.listdir(self.docs_dir):
+            if filename.endswith((".txt", ".md")):
+                file_path = os.path.join(self.docs_dir, filename)
+                with open(file_path, "r", encoding="utf-8") as file:
+                    content = file.read()
+                    document = Document(
+                        document_id=filename,
+                        content=content,
+                        mime_type="text/plain",
+                        metadata={"filename": filename},
+                    )
+                    documents.append(document)
+
+        if documents:
+            self.client.memory.insert(
+                bank_id=self.memory_bank_id,
+                documents=documents,
+            )
+            print(f"Loaded {len(documents)} documents from {self.docs_dir}")
+
+    async def initialize_agent(self):
+        """Initialize the agent with model registration and configuration."""
+
+        if "1b" in MODEL_NAME:
+            model_name = "Llama3.2-1B-Instruct"
+        elif "3b" in MODEL_NAME:
+            model_name = "Llama3.2-3B-Instruct"
+        elif "8b" in MODEL_NAME:
+            model_name = "Llama3.1-8B-Instruct"
+        else:
+            model_name = MODEL_NAME
+
```

Comment:
we should not do any such thing since this is code that developers will see and get confused. let's just use the full HF repo names since they are now supported by the stack

Reply:
The MODEL_NAME here is actually Ollama's model name which is difference than our meta MODEL_name, eg llama3.2:1b-instruct-fp16: "Llama3.2-1B-Instruct", I wonder if there is a easy way to do the model name mapping.